pkg/agent/openflow/pipeline.go

```diff
@@ -1228,6 +1376,40 @@ func (c *client) l3FwdFlowToRemoteViaGW(
 		Done()
 }
 
+// l3FwdServiceDefaultFlowsViaGW generates the default L3 forward flow to support Service traffic to pass through Antrea gateway.
+func (c *client) l3FwdServiceDefaultFlowsViaGW(ipProto binding.Protocol, category cookie.Category) []binding.Flow {
+	gatewayMAC := c.nodeConfig.GatewayConfig.MAC
+
+	flows := []binding.Flow{
+		/* This flow is used to match the packets of Service traffic:
+			- NodePort/LoadBalancer request packets which pass through Antrea gateway and the Service Endpoint is on host network.
+			- ClusterIP request packets which are from Antrea gateway and the Service Endpoint is on host network.
+		  The matched packets should leave through Antrea gateway, however, they also enter through Antrea gateway. This
+		  is hairpin traffic.
+		*/
+		c.pipeline[l3ForwardingTable].BuildFlow(priorityLow).MatchProtocol(ipProto).
+			MatchCTMark(ServiceCTMark, nil).
+			MatchCTStateRpl(false).
+			MatchCTStateTrk(true).
+			MatchRegRange(int(marksReg), macRewriteMark, macRewriteMarkRange).
+			Action().SetDstMAC(gatewayMAC).
+			Action().ResubmitToTable(l3DecTTLTable).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done(),
+		// This flow is used to match the response packets of NodePort/LoadBalancer traffic. The destination MAC address
+		// and output port will be set on serviceResponseProcessTable.
```

Comment:
Didn't find `serviceResponseProcessTable`, please refer to correct names, otherwise hard to track how the pipeline is connected.

Reply:
I forgot deleting this one.